app/components/containerLogs/containerLogsController.js

```diff
@@ -1,75 +1,84 @@
 
-  $('#loadingViewSpinner').show();
-  Container.get({id: $transition$.params().id}, function (d) {
-    $scope.container = d;
-    $('#loadingViewSpinner').hide();
-  }, function (e) {
-    $('#loadingViewSpinner').hide();
-    Notifications.error('Failure', e, 'Unable to retrieve container info');
-  });
-
   function getLogs() {
     $('#loadingViewSpinner').show();
     getLogsStdout();
     getLogsStderr();
     $('#loadingViewSpinner').hide();
   }
 
+  function parseLogResults(data) {
+    // Replace carriage returns with newlines to clean up output
+    data = data.replace(/[\r]/g, '\n');
+    // Strip 8 byte header from each line of output
+    data = data.substring(8);
+    data = data.replace(/\n(.{8})/g, '\n');
+    return data;
+  }
+
   function getLogsStderr() {
-    ContainerLogs.get($transition$.params().id, {
+    ContainerService.logs({
+      id: $transition$.params().id,
       stdout: 0,
       stderr: 1,
-      timestamps: $scope.state.displayTimestampsErr,
+      timestamps: $scope.state.displayTimestampsOut,
       tail: $scope.tailLines
-    }, function (data, status, headers, config) {
-      // Replace carriage returns with newlines to clean up output
-      data = data.replace(/[\r]/g, '\n');
-      // Strip 8 byte header from each line of output
-      data = data.substring(8);
-      data = data.replace(/\n(.{8})/g, '\n');
-      $scope.stderr = data;
+     }).then(function(data) {
+      $scope.stderr = parseLogResults(data);
     });
   }
 
   function getLogsStdout() {
-    ContainerLogs.get($transition$.params().id, {
+    ContainerService.logs({
+      id: $transition$.params().id,
       stdout: 1,
       stderr: 0,
       timestamps: $scope.state.displayTimestampsOut,
       tail: $scope.tailLines
-    }, function (data, status, headers, config) {
-      // Replace carriage returns with newlines to clean up output
-      data = data.replace(/[\r]/g, '\n');
-      // Strip 8 byte header from each line of output
-      data = data.substring(8);
-      data = data.replace(/\n(.{8})/g, '\n');
-      $scope.stdout = data;
+    }).then(function(data) {
+      $scope.stdout = parseLogResults(data);
     });
   }
 
-  // initial call
-  getLogs();
-  var logIntervalId = window.setInterval(getLogs, 5000);
+  function getContainer() {
+    $('#loadingViewSpinner').show();
+    ContainerService.container($transition$.params().id).then(function(container) {
+      $scope.container = container;
+    }).catch(function(err) {
+      Notifications.error('Failure', err, 'Unable to retrieve container info');
+    }).finally(function() {
+      $('#loadingViewSpinner').hide();
+    });
+  }
 
-  $scope.$on('$destroy', function () {
-    // clearing interval when view changes
-    clearInterval(logIntervalId);
-  });
+  function initView() {
+    getContainer();
+    getLogs();
 
-  $scope.toggleTimestampsOut = function () {
-    getLogsStdout();
-  };
+    var logIntervalId = window.setInterval(getLogs, 5000);
```

Comment:
I suggest to use [$interval](https://docs.angularjs.org/api/ng/service/$interval) service instead of native `setInterval` mostly because `setInterval` is an asynchronous action which makes harder for AngularJS to track scope changes in it, $interval is basically a `setInterval` with applied scope so you don't need to worry about repaints and value changes.

Reply:
Thanks, I will use that.